doc/modules/mod_amp.md

```diff
@@ -74,3 +74,56 @@ mod_amp supports.
 implemented. It follows that `mod_amp`, in its current state, should only be
 enabled for servers/domains where user presence leaks are not a threat, i.e
 services where all users can see each other's presence by default.
+
+### Modifications
+
+The following behaviour differs from or extends the guidelines provided in the XEP.
+
+* The action for the `deliver` condition with value `stored` is deferred until the message is stored by `mod_mam` or `mod_offline`.
+* The action for the `deliver` condition with value `direct` is deferred until the message is sent to the recipient's socket.
+
+### Server Processing Details
+
+When a message with AMP rules is being processed by the server, several system events may occur. For a given event, the rules are processed and each of them can get the *matched* or *undecided* status or, if the conditions are not met, it gets no status. If any rules get the *matched* status, the action for the first of them is performed. After that, the rule list is filtered so that only the *undecided* ones are left in the message, as they may be matched later.
```

Comment:
We try to follow `one sentence per line` rule, which makes future editing and comparing easier. It's not sth written on stone though.

Reply:
I will respect that rule. Seems a bit counter intuitive (lines become one chunk of text anyway) but it can minimize the scope of future changes indeed. The problem with `one sentence per line` is that it often encourages rewording the text just to have fitting line lengths. The content sometimes suffers during the process :)

Reply:
A single sentence may be longer than the line limit IMO. More important is to split the line after `.`

Reply:
Of course it may be longer, but then it is tempting to shorten sentences just to make them easily editable :)